### PR TITLE
261 wire pledge modal

### PR DESCRIPTION
**Pull request by charleshuang80**
I rule and have completed some work on Case Manager that's ready for review! @colinxfleming this PR still needs some work, but I know that you were planning on reviewing some things over the weekend, so I wanted you to be able to get a quick look and see where this is headed.

This pull request makes the following changes:
* makes the pledge modal screens show/hide work (still needs some tweaks but basics are there)
* moves the pledge modal code from application.js to pledges.coffee

It relates to the following issue #s: 
* #261

Things that still need to get done:
* A few TODOs in comments in pledges.coffee
* Hook up the pledge modal Finish/submit button
* clean up commented out code

Questions:
* I started to add some of the information for the pledge (i.e. clinic info), but that information was taken out of the Abortion Information partial. I'm not sure what the plan is for that, so should I leave that out for now?
* Also need to know if I should leave the $100 as is
* Was there a plan to generate (and download) the pdf for the pledge, or is how that should work totally open?

I'll continue to make this all work but totally feel free to jump in and change/add anything.

### CODE REVIEW

File: app/views/pregnancies/_abortion_information.html.erb [hunk 1]
```diff
@@ -11,7 +11,7 @@
           <div class="info-form-left">
             <%= f.fields_for (pregnancy.clinic || pregnancy.build_clinic) do |cl| %>
               <h4>Clinic details</h4>
-              <%= cl.select :name, options_for_select(clinic_options, pregnancy.clinic.name), label: 'Clinic name', autocmomplete: 'off' %>
+              <%= cl.select :name, options_for_select(clinic_options, pregnancy.clinic.name), label: 'Clinic name', autocomplete: 'off' %>
```
Comment (colinxfleming) [inline app/views/pregnancies/_abortion_information.html.erb]: oops

**Comment by colinxfleming** [comment]
Hey @charleshuang80 , thanks for yr hustle on this.

Down the line on your questions:
* *I started to add some of the information for the pledge (i.e. clinic info), but that information was taken out of the Abortion Information partial. I'm not sure what the plan is for that, so should I leave that out for now?* UX testing revealed that there was no need for these as inputs so we removed. For this dry run let's not worry about the `Clinic Location` field. 
* *Also need to know if I should leave the $100 as is* Yeah, don't worry about this for now, placeholder is fine.
* *Was there a plan to generate (and download) the pdf for the pledge, or is how that should work totally open?* Saving that functionality as a release 2 feature, probably. 

It looks like the js stuff is starting to cook. Do you expect you'll be able to wire up that checkbox before Sunday morning or so? Please let me know if not and I'll pick it up and run with it -- totally cool either way!

File: app/assets/javascripts/pledges.coffee [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+    $ps2.hide()
+    $ps3.show()
+    console.log 'pledge continue'
+
+  $('.pledge_finish').on 'click', ->
+    $ps3.hide()
+    console.log 'pledge finish'
+
+  # TODO
+  # hook up go back button
```
Comment (colinxfleming) [inline app/assets/javascripts/pledges.coffee]: Let's add this as a backlog issue and skip for MVP

File: app/assets/javascripts/pledges.coffee [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+  # do resetScreens on ready/load?
+  # on close, reset the screens so if modal opens again, will show screen 1 properly
```
Comment (colinxfleming) [inline app/assets/javascripts/pledges.coffee]: yes pls

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -19,34 +19,32 @@
...
-	$('.pledge_back_review').click(function () {
-		$('.pledge_modal_screen3').hide();
-		$('.pledge_modal_screen2').show();		
-	});
-
-	$('.pledge_finish').click(function () {
-		$('.pledge_modal_screen3').hide();		
-	});
-
+//
```
Comment (colinxfleming) [inline app/assets/javascripts/application.js]: Once you get the coffeescript stuff working feel free to nuke all this

**Comment by charleshuang80** [comment]
@colinxfleming, for the pledge modal, added functionality for the go back links and resetting the screens on close to work. The form is added, and will hit the controller, but none of the params are hooked up for the form. I have to stop for now, though. If you get to it before me tomorrow that is great, but I will hook it up if not.

**Pull request merged**